extensions/kafka-streams/runtime/src/main/java/io/quarkus/kafka/streams/runtime/health/KafkaStreamsTopicsHealthCheck.java

```diff
@@ -10,18 +10,19 @@
 import jakarta.enterprise.context.ApplicationScoped;
 import jakarta.inject.Inject;
 
-import org.eclipse.microprofile.health.HealthCheck;
 import org.eclipse.microprofile.health.HealthCheckResponse;
 import org.eclipse.microprofile.health.HealthCheckResponseBuilder;
 import org.eclipse.microprofile.health.Readiness;
 import org.jboss.logging.Logger;
 
 import io.quarkus.kafka.streams.runtime.KafkaStreamsRuntimeConfig;
 import io.quarkus.kafka.streams.runtime.KafkaStreamsTopologyManager;
+import io.smallrye.health.api.AsyncHealthCheck;
+import io.smallrye.mutiny.Uni;
 
 @Readiness
 @ApplicationScoped
-public class KafkaStreamsTopicsHealthCheck implements HealthCheck {
+public class KafkaStreamsTopicsHealthCheck implements AsyncHealthCheck {
```

Comment:
I actually don't think this is going to work properly because `KafkaStreamsTopicsHealthCheck` makes a blocking call to `listTopics`

Reply:
You're right of course, I guess the `Uni.createFrom().item` should enclose the entire method body to make it asynchronous. Pushed the changes (+ squashed).

Reply:
Thanks. I'll let @ozangunalp take it from here

Reply:
This is not as easy as it seems. We do not have a reactive Kafka Admin client checking required topics on the broker. So this health check is inherently blocking. Wrapping it in Uni doesn't make it reactive, unfortunately. 

And even though we'd transformed it to reactive I don't think it'll resolve the referenced issue.

We may discuss however, to move the Kafka Streams topics check from readiness to startup. That would help with the pods getting killed.

Reply:
@ozangunalp  Thanks for your input, so we simple replace `@Readiness` with `@Startup` in `KafkaStreamsTopicsHealthCheck`? What about the `KafkaStreamsStateHealthCheck` annotated with `@Liveness`?